src/query/api/v1/handler/prometheus/common.go

```diff
@@ -346,56 +349,11 @@ func RenderTagValuesResultsJSON(
 	return jw.Close()
 }
 
-type tag struct {
-	name  string
-	value string
-}
-
-func writeTagsHelper(
-	jw *json.Writer,
-	completedTags []storage.CompletedTag,
-	tags []tag,
-) {
-	if len(completedTags) == 0 {
-		jw.BeginObject()
-		for _, tag := range tags {
-			jw.BeginObjectField(tag.name)
-			jw.WriteString(tag.value)
-		}
-
-		jw.EndObject()
-		return
-	}
-
-	firstResult := completedTags[0]
-	name := string(firstResult.Name)
-
-	copiedTags := make([]tag, len(tags)+1)
-	copy(copiedTags, tags)
-	for _, value := range firstResult.Values {
-		copiedTags[len(tags)] = tag{name: name, value: string(value)}
-		writeTagsHelper(jw, completedTags[1:], copiedTags)
-	}
-}
-
-func writeTags(
-	jw *json.Writer,
-	results []*storage.CompleteTagsResult,
-) {
-	for _, result := range results {
-		jw.BeginArray()
-		tags := result.CompletedTags
-		if len(tags) > 0 {
-			writeTagsHelper(jw, result.CompletedTags, nil)
-		}
-		jw.EndArray()
-	}
-}
-
-// RenderSeriesMatchResultsJSON renders series match results to json format
+// RenderSeriesMatchResultsJSON renders series match results to json format.
 func RenderSeriesMatchResultsJSON(
 	w io.Writer,
-	results []*storage.CompleteTagsResult,
+	results []models.Metrics,
+	dropRole bool,
```

Comment:
Where is this coming from? Config?

Reply:
Just auto-enabled for all prom requests, and auto-disabled for graphite.

Reply:
Okay cool - maybe add a todo to config it. But at this point maybe removing some config stuff is the way to go..